lib/adagrams.rb

```diff
@@ -0,0 +1,95 @@
+require "pry"
+
+# WAVE 1
+LETTER_POOL = {a: 9, b: 2, c: 2, d: 4, e: 12, f: 2, g: 3, h: 2, i: 9, j: 1, k: 1, l: 4, m: 2,
+               n: 6, o: 8, p: 2, q: 1, r: 6, s: 4, t: 6, u: 4, v: 2, w: 2, x: 1, y: 2, z: 1}
+
+def draw_letters
+  letter_draw = LETTER_POOL.map do |key, value|
+    (key.to_s * value).split("")
+  end
+  letter_draw.flatten!
+
+  user_hand = letter_draw.sample(10)
+  return user_hand
+end
+
+# WAVE 2
+def uses_available_letters?(input, letters_in_hand)
+  uses_available_letters = true
+  input.each_char do |letter|
+    user_count = input.count(letter)
+    hand_count = letters_in_hand.count(letter)
+    if user_count > hand_count
+      uses_available_letters = false
+      break
```

Comment:
Rather than keeping this extra variable `uses_available_letters` here, you could `return false` here, and `return true` at the bottom.